File: fixtures/polyglot/scala/orders/PaymentService.scala
package com.acme.scalaorders

trait PaymentPort {
  def charge(request: PaymentRequest): PaymentReceipt
}

final class PaymentRequest(val orderId: String) {
}

final class PaymentReceipt(val id: String) {
}

final class PaymentService(invoices: InvoiceRepository) extends PaymentPort {
  def charge(request: PaymentRequest): PaymentReceipt = invoices.create(request)
}

final class InvoiceRepository {
  def create(request: PaymentRequest): PaymentReceipt = PaymentReceipt(request.orderId)
}

File: fixtures/polyglot/csharp/Orders/PaymentService.cs
namespace Acme.Orders;

public interface IPaymentPort
{
    PaymentReceipt Charge(PaymentRequest request);
}

public sealed class PaymentService : IPaymentPort
{
    private readonly InvoiceRepository invoices;

    public PaymentService(InvoiceRepository invoices)
    {
        this.invoices = invoices;
    }

    public PaymentReceipt Charge(PaymentRequest request)
    {
        return invoices.Create(request);
    }
}

public sealed record PaymentReceipt(string Id);
public sealed record PaymentRequest(string OrderId);
public sealed class InvoiceRepository
{
    public PaymentReceipt Create(PaymentRequest request) => new PaymentReceipt(request.OrderId);
}
